### ISSUE TITLE
[MD049] Incorrectly flagged

### ISSUE THREAD

**Issue by skyzyx** [rvben/rumdl#681]
### rumdl version

0.2.20

### Installation method

Homebrew

### Bug description

For `MD049`, I've configured the rule to use _underscores_ for emphasis. This bullet point is getting flagged.

> * If there are multiple bulletpoints that are extremely similar, try to consolidate them into a single bulletpoint that captures the overall change. For example, if there are multiple bulletpoints about adding logging to different functions, you could consolidate them into a single bulletpoint like "* Added logging to key functions for better observability."

<img width="1540" height="222" alt="Image" src="https://github.com/user-attachments/assets/9ac7d5b2-0997-497b-97c9-28d587e44295" />

### Steps to reproduce

1. Install the latest version of the `rumdl` CLI tool.
2. Install the latest version of the `rumdl` VSCode extension (I'm using Kiro, but VSCode has the same issue).
3. Define `[MD049]` in a `rumdl.toml` file at the root of the project.
4. Add the aforementioned text to a new Markdown document.

### Expected behavior

There should be no error/warning.

### Actual behavior

There is an error/warning.

### Configuration

```toml
# https://rumdl.dev/md049/
[MD049]
  # emphasis-style
  enabled = true
  style = "underscore"
```

### Operating system

macOS

### Additional context

The Commonmark "dingus" shows that these are expected to NOT be emphasis markers — particularly if you check the AST.

https://spec.commonmark.org/dingus/?text=*%20If%20there%20are%20multiple%20bulletpoints%20that%20are%20extremely%20similar%2C%20try%20to%20consolidate%20them%20into%20a%20single%20bulletpoint%20that%20captures%20the%20overall%20change.%20For%20example%2C%20if%20there%20are%20multiple%20bulletpoints%20about%20adding%20logging%20to%20different%20functions%2C%20you%20could%20consolidate%20them%20into%20a%20single%20bulletpoint%20like%20%22*%20Added%20logging%20to%20key%20functions%20for%20better%20observability.%22

**Comment by rvben** [comment]
Thanks for the clear report and the CommonMark dingus link, @skyzyx.

Root cause: MD049 was treating the leading `*` list marker as an emphasis delimiter and pairing it with the stray `*` inside the quoted `"* Added ..."` text, so the whole line looked like one emphasis span. Per CommonMark a `*` followed by whitespace can't open emphasis, so there was nothing to flag.

Fix: MD049 now applies the CommonMark left/right-flanking rule, so whitespace-flanked markers and list markers aren't mistaken for emphasis (genuine emphasis like `*foo * bar*` still works). A related MD037 false positive is fixed in the same change.

Fixed in v0.2.21.

**Comment by skyzyx** [comment]
Appreciated. Thank you!
